core/types/bal_encoding.go

```diff
@@ -0,0 +1,361 @@
+package types
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/holiman/uint256"
+)
+
+//go:generate go run github.com/ferranbt/fastssz/sszgen  --output bal_encoding_generated.go --path . --objs encodingStorageWrite,encodingStorageWrites,encodingCodeChange,encodingBalanceChange,encodingAccountNonce,encodingCodeChange,encodingAccountAccess,encodingBlockAccessList
+
+// These are objects used as input for the access list encoding. They mirror
+// the spec format.
+
+type encodingBlockAccessList struct {
+	Accesses []encodingAccountAccess `ssz-max:"300000"`
+}
+
+// toBlockAccessList converts out of the encoding format, returning an error if
+// values in the encoder object are not properly ordered according to the spec.
+func (e *encodingBlockAccessList) toBlockAccessList() (*BlockAccessList, error) {
+	res := NewBlockAccessList()
+	var prevAccount *common.Address
+	for _, encAccountAccess := range e.Accesses {
+		if prevAccount != nil {
+			if bytes.Compare(encAccountAccess.Address[:], (*prevAccount)[:]) <= 0 {
+				return nil, fmt.Errorf("block access list accounts not in lexicographic order")
+			}
+		}
+		aa, err := encAccountAccess.toAccountAccess()
+		if err != nil {
+			return nil, err
+		}
+		res.accounts[encAccountAccess.Address] = aa
+	}
+	return &res, nil
+}
+
+type encodingCodeChange struct {
+	TxIndex uint16
+	Code    []byte `ssz-max:"24576"`
+}
+
+type encodingAccountAccess struct {
+	Address        [20]byte                `ssz-size:"20"`
+	StorageWrites  []encodingSlotWrites    `ssz-max:"300000"`
+	StorageReads   [][32]byte              `ssz-max:"300000"`
+	BalanceChanges []encodingBalanceChange `ssz-max:"300000"`
+	NonceChanges   []encodingAccountNonce  `ssz-max:"300000"`
+	Code           []encodingCodeChange    `ssz-max:"1"`
```

Comment:
Also needs to have a max of 30_000 according to spec (maybe the spec should be updated here)

Reply:
I think this is probably a typo in the spec (I will PR a change to the EIP).  The max should be 1:  after selfdestruct removal, the code of an account can only change once (we don't include delegations in the BAL because they can be determined statically by looking at the block txs).